_posts/2017-10-04-Law-of-quality.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: post
+title: The Law of Three for Software Quality
+author: lokori
+excerpt: There is a simple rule-of-thumb for software quality which seems to remain true over the years: 
+If the system properly handles dates, characters and money, it has good quality. 
+Short of scientific evidence, empirical arguments are provided. Try and see how your system measures up against this law.
+tags:
+- quality
+- bugs
+- encoding
+- SQL
+- Java
+---
+
+## The Law of Three
+
+This is the law. 
+
+### If the system properly handles dates, characters and money, it has good quality.
+
+I have been talking about it every now and then for quite some time. It seems to hold true in the world of professional software development where I work as years pass by. Most of the real world systems handle characters and dates in some fashion, and getting these two right is a major challenge even if money is not involved. The law sets a sufficient condition; I rarely see a software system of non-trivial size which got all these right. 
+
+This post examines the difficulties in these three matters to shed some light into why it's actually quite difficult. The intention is not deal with these subjects a in thorough manner - that would fill a book I'm not qualified to write.
```

Comment:
“why it's actually” -> “why they're actually” maybe?

“not deal” -> “not to deal”

“-” “—” (👓)